### ISSUE TITLE
Define syllabus

### ISSUE THREAD

**Issue by mofeing** [mofeing/computational-complexity#2]
The main problem I see following the course is that I do not see a clear structure of the contents and the published syllabus is not adjusting at what we are seeing in class. Also, a lot of different teachings overlap between lectures, blurring the line between sections.

I am beginner in this subject and some background is required in a master, so my complaint may not be very valid. But knowing what comes after each content and packaging concepts helps to build a relation between in the mind.

We should define a "Table of Contents" based from class notes before distributing the work between us.

This is my propose:

1. Models of Computation
    1. The (Deterministic) Turing Machine
    2. The Universal Turing Machine
        - The Church-Turing thesis
    3. Complexity classes
2. The Time and Space Hierarchy theorems

The list will be updated as the course continues.

**Comment by jproyo** [comment]
This is my purpose. It follows more or less the order of the classes

https://github.com/mofeing/computational-complexity/wiki/Syllabus

**Comment by mofeing** [comment]
@jproyo I like it, but I suggest to fuse some chapters that are related:
- **Encodings** into **The (Deterministic) Turing Machine**: They are the information TMs work with.
- **The Universal Turing Machine with Time Bounds** into **The Time Hierarchy Theorem**: As long as I understood, the _time bounds_ is basically a counter built into a TM for measuring the time complexity of an algorithm.
- **Cook's Theorem**, **SAT reduces to CLIQUE**, **3-SAT is NP-complete**, **E3-SAT is NP-complete**... look like a whole big chapter about SAT/NP-completeness. It fits with the unit 2 of the official syllabus of the course.

**Comment by jproyo** [comment]
> @jproyo I like it, but I suggest to fuse some chapters that are related:
> 
> * **Encodings** into **The (Deterministic) Turing Machine**: They are the information TMs work with.

The problem to fuse `encodings` there is that it is not only for TM but also for NTM. Apart from that is like a premilinary thing to work with TM and NTM

> * **The Universal Turing Machine with Time Bounds** into **The Time Hierarchy Theorem**: As long as I understood, the _time bounds_ is basically a counter built into a TM for measuring the time complexity of an algorithm.

I dont think so. Universal TM is more a general concept. It is used for doing the proves in Time Hierarchy Theorems, but also in other problems or theorems, such as Halting Problem

> * **Asymptotic Grow Rates** into **Complexity Classes**: Big-O, small-o notations are just the mathematical notation for specifying the complexity classes.

👍 

> * **Cook's Theorem**, **SAT reduces to CLIQUE**, **3-SAT is NP-complete**, **E3-SAT is NP-complete**... look like a whole big chapter about SAT/NP-completeness. It fits with the unit 2 of the official syllabus of the course.

👍

**Comment by mofeing** [comment]
> The problem to fuse `encodings` there is that it is not only for TM but also for NTM. Apart from that is like a premilinary thing to work with TM and NTM

Yeah, but the first chapter is kind of introductive and encodings should be explained there as the way TMs "encode"/work with information. NTM is a variation of TM, but the way both use encodings is the same.

> I dont think so. Universal TM is more a general concept. It is used for doing the proves in Time Hierarchy Theorems, but also in other problems or theorems, such as Halting Problem

Yeah yeah, I didn't mean Universal TM generically but Universal TM _with time bounds_ specifically.

BTW I've reflected the other two changes into the syllabus wiki.
